hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/utils/SparkMetadataWriterUtils.java

```diff
@@ -328,15 +347,16 @@ public static ExpressionIndexComputationMetadata getExprIndexRecords(List<Pair<S
 
     // Apply expression index and generate the column to index
     HoodieExpressionIndex<Column, Column> expressionIndex =
-        new HoodieSparkExpressionIndex(indexDefinition.getIndexName(), indexDefinition.getIndexFunction(), indexDefinition.getSourceFields(), indexDefinition.getIndexOptions());
+        new HoodieSparkExpressionIndex(indexDefinition.getIndexName(), indexDefinition.getIndexFunction(), indexDefinition.getIndexType(), indexDefinition.getSourceFields(),
+            indexDefinition.getIndexOptions());
```

Comment:
Any reason of not passing in the `indexDefinition` as a whole to the constructor?

Reply:
Refactored.